### PR TITLE
🔥 add firestore recipe 🔥

### PR DESCRIPTION
**Pull request by kevinzunigacuellar**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Add firestore 🔥 recipe 🧑‍🍳

### CODE REVIEW

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 1]
```diff
@@ -547,6 +551,319 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
 
+## Connecting to Firestore database
+
+### Prerequisites
+
+- An Astro project initialized with Firebase as described in the [Initializing Firebase in Astro](#initializing-firebase-in-astro) section.
+
+- A Firebase project with a Firestore database. You can follow the [Firebase documentation](https://firebase.google.com/docs/firestore/quickstart) to create a new project and set up a Firestore database. 
+
+Our Firestore collection will be called **friends** and will contain documents with the following fields:
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:562]:
```suggestion
In this recipe, the Firestore collection will be called **friends** and will contain documents with the following fields:
```

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 2]
```diff
@@ -547,6 +551,319 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+
+- `id`: autogenerated by Firestore
+- `name`: a string field
+- `age`: a number field
+- `isBestFriend`: a boolean field
+
+Feel free to follow this example or adapt it to your own Firestore collection.
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:569]:
```suggestion
```

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 3]
```diff
@@ -547,6 +551,319 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+### Creating the server endpoints
+
+We will create two endpoints to interact with the Firestore database:
+
+- `POST /api/friends`: to create a new document in the friends collection.
+- `GET /api/friends/:id`: to get a document in the friends collection.
+- `POST /api/friends/:id`: to update a document in the friends collection.
+- `DELETE /api/friends/:id`: to delete a document in the friends collection.
+
+Create two new files in a new directory `src/pages/api/friends/`: `index.ts` and `[id].ts`.
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:580]:
```suggestion
Create two new files in a new directory `src/pages/api/friends/`: `index.ts` and `[id].ts`. These will create two server endpoints to interact with the Firestore database in the following ways:

- `POST /api/friends`: to create a new document in the friends collection.
- `GET /api/friends/:id`: to get a document in the friends collection.
- `POST /api/friends/:id`: to update a document in the friends collection.
- `DELETE /api/friends/:id`: to delete a document in the friends collection.


```

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 4]
```diff
@@ -547,6 +551,319 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+        - **index.ts**
+        - **[id].ts**
+- .env
+- astro.config.mjs
+- package.json
+</FileTree>
+
+### Creating pages
+
+Now that we have Firestore endpoints set up, we can create the pages that will use them.
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:699]:
```suggestion
Create the pages that will use the Firestore endpoints:
```

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 5]
```diff
@@ -547,6 +551,319 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+### Creating pages
+
+Now that we have Firestore endpoints set up, we can create the pages that will use them. 
+
+- `src/pages/add.astro` - will contain a form to add a new friend.
+- `src/pages/edit/[id].ts` - will contain a form to edit a friend and a button to delete a friend.
+- `src/pages/friend/[id].ts` - will contain the details of a friend.
+- `src/pages/dashboard.astro` - will display a list of friends.
+
+The example `src/pages/add.astro` below includes a form that will send a `POST` request to the `/api/friends` endpoint. This endpoint will create a new friend using the data from the form and then will redirect the user to the `/dashboard` page.
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:706]:
```suggestion
#### Add a new record

The example `src/pages/add.astro` below includes a form that will send a `POST` request to the `/api/friends` endpoint. This endpoint will create a new friend using the data from the form and then will redirect the user to the `/dashboard` page.
```

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 6]
```diff
@@ -547,6 +551,319 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+    <label for="age">Age</label>
+    <input type="number" id="age" name="age" />
+    <label for="isBestFriend">Is best friend?</label>
+    <input type="checkbox" id="isBestFriend" name="isBestFriend" />
+    <button type="submit">Add friend</button>
+  </form>
+</Layout>
+```
+
+`src/pages/edit/[id].astro` will contain a form to edit a friend data and a button to delete a friend. On submit, this page will send a `POST` request to the `/api/friends/:id` endpoint to update a friend data.
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:727]:
```suggestion
#### Edit or Delete a record

`src/pages/edit/[id].astro` will contain a form to edit a friend data and a button to delete a friend. On submit, this page will send a `POST` request to the `/api/friends/:id` endpoint to update a friend data. 
```

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 7]
```diff
@@ -547,6 +551,319 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+const friend = friendSnapshot.data() as Friend
+---
+<Layout title={friend.name}>
+  <h1>{friend.name}</h1>
+  <p>Age: {friend.age}</p>
+  <p>Is best friend: {friend.isBestFriend ? 'Yes' : 'No'}</p>
+</Layout>
+```
+
+Finally, `src/pages/dashboard.astro` will display a list of friends. Each friend will have a link to their details page and an edit button that will redirect the user to the edit page.
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:824]:
```suggestion
#### Display records with an edit button

Finally, `src/pages/dashboard.astro` will display a list of friends. Each friend will have a link to their details page and an edit button that will redirect the user to the edit page.
```

**Review by sarah11918 — COMMENTED** [review]
Amazing work, Kevin! so thrilled to have this!  I'm just straight ahead committing some mostly formatting changes, and I'll let others read over to see whether they catch anything, and this is well in time for today!  :partying_face:

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 8]
```diff
@@ -547,6 +551,324 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+    <label for="age">Age</label>
+    <input type="number" id="age" name="age" />
+    <label for="isBestFriend">Is best friend?</label>
+    <input type="checkbox" id="isBestFriend" name="isBestFriend" />
+    <button type="submit">Add friend</button>
+  </form>
+</Layout>
+```
+
+#### Edit or Delete a record
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:728]:
```suggestion
#### Edit a record
```

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 9]
```diff
@@ -547,6 +551,324 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+      method: 'DELETE'
+    })
+    if (response.redirected) {
+      window.location.assign(response.url)
+    }
+  })
+</script>
+```
+
+`src/pages/friend/[id].astro` will display the details of a friend.
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:790]:
```suggestion
#### Display an individual record

`src/pages/friend/[id].astro` will display the details of a friend.
```

File: src/content/docs/en/guides/backend/google-firebase.mdx [hunk 10]
```diff
@@ -547,6 +551,326 @@ When clicked, the Google sign in button will open a popup window to sign in with
...
+const friend = friendSnapshot.data() as Friend
+---
+<Layout title={friend.name}>
+  <h1>{friend.name}</h1>
+  <p>Age: {friend.age}</p>
+  <p>Is best friend: {friend.isBestFriend ? 'Yes' : 'No'}</p>
+</Layout>
+```
+
+#### Display records with an edit button
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/google-firebase.mdx:829]:
```suggestion
#### Display a list of records with an edit button
```

**Review by sarah11918 — APPROVED** [review]
Leaving my approval, and we can summon team docs to PR any typo fixes they spot in prod.

Happy for you to merge when you're happy, Kevin!  :partying_face:

**Comment by kevinzunigacuellar** [comment]
> Leaving my approval, and we can summon team docs to PR any typo fixes they spot in prod.
> 
> Happy for you to merge when you're happy, Kevin! 🥳

Yolo pushing

**Pull request merged**